tests/pos/T2349.hs

```diff
@@ -0,0 +1,17 @@
+{-# LANGUAGE GADTs #-}
+{-@ LIQUID "--reflection" @-}
+module TestModule where
+
+data Expr t where
+  I :: Int -> Expr Int
+
+  NIL :: Expr [t]
+  CONS :: Int -> Expr t -> Expr [t] -> Expr [t]
+
+{-@
+data Expr t where
+  I :: Int -> Expr Int
+  NIL :: Expr {v:[t] | len v = 0}
+  CONS :: xxxxx:_ -> head:Expr t -> tail:Expr {v:[t] | len v = xxxxx} ->
+          Expr {v:[t] | len v = xxxxx+1}
```

Comment:
Should the test include a use of CONS? e.g.

```Haskell
{-@
fail e
e :: Expr {v:[Int] | len v = 0}
@-}
e :: Expr [Int]
e = CONS 0 (I 2) NIL
```

Reply:
Edited! (I added a negative test instead)